Move tracking docs to JSDoc

tomalec opened the pull request:
_This is part 2 of https://github.com/woocommerce/google-listings-and-ads/pull/1369_
### Changes proposed in this Pull Request:


1. Move Tracking events' documentation closer to the source, to JSDoc next to emitting code. So the documentation is easier to maintain and introduces less context switching for devs.
3. Format docs for PHP-triggered in the same convention as JS ones. https://github.com/woocommerce/google-listings-and-ads/pull/1401/commits/6df63f328a51bf1a908714e10d96b1076632af93
4. I also changed the event names to contain `gla_` prefix to follow the Track conversions guideline: 
   > They should be hard coded so they are easily searchable in the codebase.


### Screenshots:

<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Review the https://github.com/woocommerce/google-listings-and-ads/blob/tweak/jsdoc-gen/src/Tracking/README.md

For easier code review, please refer to the [individual commits list](https://github.com/woocommerce/google-listings-and-ads/pull/1401/commits)

### Additional details:

1. **Discissed in https://github.com/woocommerce/google-listings-and-ads/pull/1369#issuecomment-1088714972** :scroll: :grey_question:  I wonder about the purpose of a few functions in `js/src/utils/recordEvent.js` do we really need those that are just 1:1 proxies? (//cc @ecgan as the first author)  like: 
   ```js
   export const recordFilterEvent = ( data ) => {
	   recordEvent( 'gla_filter', data );
   };
   ```
   - I think having them here leads to obsolete code, like `recordPreLaunchChecklistCompleteEvent` which is still there, the event is still in docs even though, it's not executed anywhere
   - I think, it's also against the principle behind: "They should be hardcoded so they are easily searchable in the codebase." - so that we should have the event name close to the trigger point in the code, not far behind proxies, and libraries.
   - I believe they blur the picture of the actual usage, like: `export const recordSetupAdsEvent = ( target, trigger = 'click' ) => {` may give an impression, that there are many emitters of that event, and that `trigger` may have multiple values. Where in fact, there is [only one emitter, and one possible combination](https://github.com/woocommerce/google-listings-and-ads/blob/0e80cf865ac1b0535c28deee10c4fd89ee679b40/js/src/setup-ads/top-bar/index.js#L19) of values which is `'back', 'click'`

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Tweak - Move Tracking events docs to JSDoc.

tomalec committed:
Move `gla_contact_information_save_button_click` docs to JSDoc.

tomalec committed:
Move `gla_datepicker_update` docs to JSDoc.

tomalec committed:
Move `gla_dashboard_edit_program_click` docs to JSDoc.

tomalec committed:
Move `gla_disconnected_accounts` docs to JSDoc.

tomalec committed:
Move `gla_documentation_link_click` docs to JSDoc.

tomalec committed:
Move `gla_edit_mc_phone_number` docs to JSDoc.

tomalec committed:
Move `gla_edit_product_click` docs to JSDoc.

tomalec committed:
Move `gla_edit_product_issue_click` docs to JSDoc.

tomalec committed:
Move `gla_filter` docs to JSDoc.

tomalec committed:
Move `gla_free_ad_credit_country_click` docs to JSDoc.

tomalec committed:
Move `gla_free_campaign_edited` docs to JSDoc.

tomalec committed:
Move `get_started_faq` docs to JSDoc.

tomalec committed:
Move `gla_get_started_notice_link_click` docs to JSDoc.

tomalec committed:
Move `gla_google_account_connect_button_click` docs to JSDoc.

tomalec committed:
Move `gla_google_account_connect_different_account_button_click` docs to JSDoc.

tomalec committed:
Move `gla_google_ads_account_link_click` docs to JSDoc.

tomalec committed:
Move `gla_google_mc_link_click` docs to JSDoc.

tomalec committed:
Move `gla_help_click` docs to JSDoc.

tomalec committed:
Move `gla_launch_paid_campaign_button_click` docs to JSDoc, remove `audience` prop,

as it was never set.

tomalec committed:
Move `gla_mc_account_connect_button_click` docs to JSDoc.

tomalec committed:
Move `gla_mc_account_connect_different_account_button_click` docs to JSDoc.

tomalec committed:
Move `gla_mc_account_create_button_click` docs to JSDoc.

tomalec committed:
Move `gla_mc_account_reclaim_url_button_click` docs to JSDoc.

tomalec committed:
Move `gla_mc_account_switch_account_button_click` docs to JSDoc.

tomalec committed:
Move `gla_mc_account_switch_url_button_click` docs to JSDoc.

tomalec committed:
Move `gla_mc_account_warning_modal_confirm_button_click` docs to JSDoc.

tomalec committed:
Move `gla_mc_phone_number_check` docs to JSDoc.

tomalec committed:
Move `gla_mc_phone_number_edit_button_click` docs to JSDoc.

tomalec committed:
Move `gla_modal_closed` docs to JSDoc.

tomalec committed:
Move `gla_modal_content_link_click` docs to JSDoc.

tomalec committed:
Move `gla_modal_open` docs to JSDoc.

tomalec committed:
Move `gla_setup_ads` docs to JSDoc.

tomalec committed:
Move `gla_setup_ads_faq` docs to JSDoc.

tomalec committed:
Move `gla_setup_mc` docs to JSDoc.

tomalec committed:
Move `gla_setup_mc_faq` docs to JSDoc.

tomalec committed:
Move `gla_table_go_to_page` docs to JSDoc.

tomalec committed:
Move `gla_table_header_toggle` docs to JSDoc.

tomalec committed:
Move `gla_table_page_click` docs to JSDoc.

tomalec committed:
Move `gla_table_sort` docs to JSDoc.

tomalec committed:
Move `gla_tooltip_viewed` docs to JSDoc.

tomalec committed:
Move `gla_wordpress_account_connect_button_click` docs to JSDoc.

tomalec committed:
Format docs for PHP-generated track events ,

in the same convention as automatically generated JS ones.

tomalec committed:
Unify JSDoc and README docs for `gla_edit_*_store_address`.

tomalec committed:
Move `gla_ads_account_connect_button_click` docs to JSDoc.

tomalec committed:
Move `gla_ads_account_create_button_click` docs to JSDoc.

tomalec committed:
Move `gla_chart_tab_click` docs to JSDoc.

tomalec committed:
Move `gla_bulk_edit_click` docs to JSDoc.

tomalec committed:
Move `gla_ads_set_up_billing_click` docs to JSDoc.

tomalec committed:
Move `gla_add_paid_campaign_clicked` docs to JSDoc.

tomalec committed:
Merge commit '289661ad0c9d1d92077b088ff5959cd80335d3cc' into tweak/jsdoc-move-1

puntope commented on js/src/components/app-table-card/recordColumnToggleEvent.js:
```diff
@@ -3,6 +3,12 @@
 import { recordTableHeaderToggleEvent } from '.~/utils/recordEvent';
 
+/**
+ * @param {*} trackEventReportId
+ * @param {*} shown
+ * @param {*} toggled
```
❓ 
Why `{*}`?

tomalec replied:
Just added doc block to put @fires and linter complained about not mentioned params. And didn't want to dig deeper to check the types, as I'm going to clean up `record*Event` functions in the following PR as discussed in https://github.com/woocommerce/google-listings-and-ads/pull/1369#issuecomment-1088714972

puntope commented on js/src/components/contact-information/store-address-card.js:
```diff
@@ -21,12 +21,13 @@ import TrackableLink from '.~/components/trackable-link';
+ * Before `1.5.0` it was called `edit_mc_store_address`.
  *
  * @event gla_edit_wc_store_address
  * @type {Object} TrackingEvent
```
```suggestion
 * @type {Object}
```

puntope replied:
Avoiding 

`WARNING: The @type tag does not permit a description; the description will be ignored. File: store-address-card.js, line: 23`

tomalec replied:
I removed that in https://github.com/woocommerce/google-listings-and-ads/pull/1369

puntope commented on js/src/components/contact-information/store-address-card.js:
```diff
@@ -122,12 +123,13 @@ export default function StoreAddressCard() {
+ * Before `1.5.0` this name was used for tracking clicking "Edit in settings" to edit the WC address. As of `>1.5.0`, that event is now tracked as `edit_wc_store_address`.
  *
  * @event gla_edit_mc_store_address
  * @type {Object} TrackingEvent
```
```suggestion
 * @type {Object}
```

puntope replied:
Avoiding 

`WARNING: The @type tag does not permit a description; the description will be ignored. File: store-address-card.js, line: 125`

tomalec replied:
I removed that in https://github.com/woocommerce/google-listings-and-ads/pull/1369

puntope commented on js/src/components/guide-page-content/index.js:
```diff
@@ -25,13 +25,23 @@ export default function GuidePageContent( { title, children } ) {
+ * Clicking on a text link within the modal content
+ *
+ * @event gla_modal_content_link_click
+ * @property {string} context indicate which link is clicked
```
```suggestion
 * @property {string} context Indicates which link is clicked
```

puntope commented on js/src/components/guide-page-content/index.js:
```diff
@@ -25,13 +25,23 @@ export default function GuidePageContent( { title, children } ) {
+ *
+ * @event gla_modal_content_link_click
+ * @property {string} context indicate which link is clicked
+ * @property {string} href link's URL
```
```suggestion
 * @property {string} href Link's URL
```

puntope commented on js/src/components/help-icon-button.js:
```diff
@@ -9,11 +9,20 @@ import GridiconHelpOutline from 'gridicons/dist/help-outline';
+ * "Help" button is clicked.
+ *
+ * @event gla_help_click
+ * @property {string} context indicate the place where the button is located, e.g. `setup-ads`.
```
```suggestion
 * @property {string} context Indicates the place where the button is located, e.g. `setup-ads`.
```

puntope commented on js/src/components/help-popover/index.js:
```diff
@@ -11,8 +11,13 @@ import GridiconHelpOutline from 'gridicons/dist/help-outline';
+/**
+ * @param {Object} props React props
+ * @param {string} props.id
+ * @param {Array<JSX.Element>} props.children
```
```suggestion
 * @param {string} props.id The Popover’s ID 
 * @param {Array<JSX.Element>} props.children The Popover’s content
```

Missing descriptions

puntope commented on js/src/components/help-popover/recordTooltipViewedEvent.js:
```diff
@@ -3,6 +3,16 @@
+ * @property {string} id (tooltip identifier)
+ */
+/**
+ * @param {string} id
```
```suggestion
 * @param {string} id The tooltip’s identifier
```

puntope commented on js/src/components/paid-ads/faqs-section.js:
```diff
@@ -10,6 +10,17 @@ import { recordEvent } from '@woocommerce/tracks';
+ * Clicking on faq items to collapse or expand it in the Setup Ads page
+ *
+ * @event gla_setup_ads_faq
+ * @property {string} id (faq identifier)
+ * @property {string} action (`expand`|`collapse`)
```
❓  Not sure if this parenthesis are necessary

tomalec replied:
Good catch. I just copied the desciptions from the old `README.md` but I'll fix it here.

puntope commented on js/src/dashboard/all-programs-table-card/edit-program-button/edit-program-prompt-modal/index.js:
```diff
@@ -14,9 +14,22 @@ import recordEvent from '.~/utils/recordEvent';
+/**
+ * @fires gla_dashboard_edit_program_click when "Continue to edit" is clicked.
+ * @param {Object} props React props.
+ * @param {string} props.programId
+ * @param {Function} props.onRequestClose
```
```suggestion
 * @param {string} props.programId The program's identifier 
 * @param {Function} props.onRequestClose Callback function when closing the modal. 
```

puntope commented on js/src/dashboard/all-programs-table-card/edit-program-button/edit-program-prompt-modal/index.js:
```diff
@@ -14,9 +14,22 @@ import recordEvent from '.~/utils/recordEvent';
+ * @param {Object} props React props.
+ * @param {string} props.programId
+ * @param {Function} props.onRequestClose
+ * @return {JSX.Element} `AppModal` with content.
```
```suggestion
 * @return {import(".~/components/app-modal").default} `AppModal` with content.
```

tomalec replied:
Correct me if I'm wrong. I think given React does not use inheritance, the returned element is not an instance of `AppModal` but new `JSX.Elemtent` that has `AppModal` inside.

puntope commented on js/src/get-started-page/unsupported-notices/index.js:
```diff
@@ -23,7 +23,18 @@ const ExternalIcon = () => (
+ */
 
+/**
+ * @fires gla_get_started_notice_link_click with `{	context: "get-started", link_id: "supported-languages" }`
```
```suggestion
 * @fires gla_get_started_notice_link_click with `{ context: "get-started", link_id: "supported-languages" }`
```

puntope commented on js/src/get-started-page/unsupported-notices/index.js:
```diff
@@ -23,7 +23,18 @@ const ExternalIcon = () => (
+ * Clicking on a text link within the notice on the Get Started page.
+ *
+ * @event gla_get_started_notice_link_click
+ * @property {string} link_id link identifier
+ * @property {string} context indicate which link is clicked
+ * @property {string} href link's URL
```
```suggestion
 * @property {string} link_id Link identifier
 * @property {string} context Indicates which link is clicked
 * @property {string} href Link's URL
```

puntope commented on js/src/get-started-page/unsupported-notices/index.js:
```diff
@@ -67,6 +78,9 @@ const UnsupportedLanguage = () => {
 };
 
+/**
+ * @fires gla_get_started_notice_link_click with `{	context: "get-started", link_id: "supported-countries" }`
```
```suggestion
 * @fires gla_get_started_notice_link_click with `{ context: "get-started", link_id: "supported-countries" }`
```

puntope commented on js/src/product-feed/issues-table-card/index.js:
```diff
@@ -80,6 +80,19 @@ const actions = (
+ * Triggered when edit links are clicked from Issues to resolve table.
+ *
+ * @event gla_edit_product_issue_click
+ * @property {string} code issue code returned from Google
+ * @property {string} issue issue description returned from Google
```
```suggestion
 * @property {string} code Issue code returned from Google
 * @property {string} issue Issue description returned from Google
```

puntope commented on js/src/product-feed/product-feed-table-card/index.js:
```diff
@@ -37,8 +37,30 @@ const EVENT_CONTEXT = 'product-feed';
+ * Triggered when the product feed "bulk edit" functionality is being used
+ *
+ * @event gla_bulk_edit_click
+ * @property {string} context name of the table
+ * @property {number} number_of_items edit how many items
+ * @property {string} visibility_to `("sync_and_show" | "dont_sync_and_show")`
```
```suggestion
 * @property {string} context Name of the table
 * @property {number} number_of_items Edit how many items
 * @property {string} visibility_to `("sync_and_show" | "dont_sync_and_show")`
```

puntope commented on js/src/product-feed/submission-success-guide/index.js:
```diff
@@ -141,11 +141,21 @@ const handleGuideFinish = ( e ) => {
+ * A modal is opend
+ *
+ * @event gla_modal_open
+ * @property {string} context indicate which modal is opened
```
```suggestion
 * @property {string} context Indicates which modal is opened
```

puntope commented on js/src/reports/summary-section.js:
```diff
@@ -17,6 +17,14 @@ const noValidData = {
+ * Triggered when a chart tab is clicked
+ *
+ * @event gla_chart_tab_click
+ * @property {string} report name of the report (e.g. `"reports-programs" | "reports-products"`)
+ * @property {string} context metric key of the clicked tab (e.g. `"sales" | "conversions" | "clicks" | "impressions" | "spend"`).
```
```suggestion
 * @property {string} report Name of the report (e.g. `"reports-programs" | "reports-products"`)
 * @property {string} context Metric key of the clicked tab (e.g. `"sales" | "conversions" | "clicks" | "impressions" | "spend"`).
```

puntope commented on js/src/setup-ads/ads-stepper/setup-accounts/free-ad-credit/index.js:
```diff
@@ -13,6 +13,16 @@ import './index.scss';
+ * Clicking on the link to view free ad credit value by country.
+ *
+ * @event gla_free_ad_credit_country_click
+ * @property {string} context indicate which page the link is in.
```
```suggestion
 * @property {string} context Indicates which page the link is in.
```

puntope commented on js/src/setup-ads/ads-stepper/setup-accounts/google-ads-account-card/authorize-ads.js:
```diff
@@ -13,6 +13,11 @@ import AppButton from '.~/components/app-button';
 
+/**
+ * @param {Object} props React props
+ * @param {string} props.additionalScopeEmail
```
Not sure what is this `additionalScopeEmail`  for, but the description is missing.

tomalec replied:
I was not sure either. Just added doc block to put `@fires` and linter complained about not mentioned params.

puntope commented on js/src/setup-ads/ads-stepper/setup-accounts/google-ads-account-card/terms-modal/index.js:
```diff
@@ -13,8 +13,24 @@ import AppDocumentationLink from '.~/components/app-documentation-link';
+ * @fires gla_ads_account_create_button_click When agreed by clicking "Create account".
+ * @param {Object} props React props
+ * @param {Function} [props.onCreateAccount]
+ * @param {Function} [props.onRequestClose]
```
```suggestion
 * @param {Function} [props.onCreateAccount] Callback function when account is created
 * @param {Function} [props.onRequestClose] Callback function when the modal is closed
```

puntope commented on js/src/setup-ads/ads-stepper/setup-billing/billing-saved-card/index.js:
```diff
@@ -16,6 +16,18 @@ import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
+ * Clicking on a Google Ads account text link.
+ *
+ * @event gla_google_ads_account_link_click
+ * @property {string} context indicate which page / module the link is in
+ * @property {string} href where the user is redirected
+ * @property {string} link_id a unique ID for the link within the page / module
```
```suggestion
 * @property {string} context Indicates which page / module the link is in
 * @property {string} href Where the user is redirected
 * @property {string} link_id A unique ID for the link within the page / module
```

puntope commented on js/src/setup-ads/ads-stepper/setup-billing/setup-card/index.js:
```diff
@@ -15,8 +15,23 @@ import './index.scss';
+ * "Set up billing" button for Google Ads account is clicked.
+ *
+ * @event gla_ads_set_up_billing_click
+ * @property {string} context indicate the place where the button is located, e.g. `setup-ads`.
+ * @property {string} link_id a unique ID for the button within the context, e.g. `set-up-billing`.
+ * @property {string} href indicate the destination where the users is directed to.
```
```suggestion
 * @property {string} context Indicates the place where the button is located, e.g. `setup-ads`.
 * @property {string} link_id A unique ID for the button within the context, e.g. `set-up-billing`.
 * @property {string} href Indicates the destination where the users is directed to.
```

puntope commented on js/src/setup-ads/ads-stepper/setup-billing/setup-card/index.js:
```diff
@@ -15,8 +15,23 @@ import './index.scss';
+/**
+ * @fires gla_ads_set_up_billing_click with `{ context: 'setup-ads', link_id: 'set-up-billing',	href: billingUrl }`
+ * @param {Object} props React props.
+ * @param {string} props.billingUrl
+ * @param {Function} props.onSetupComplete
```
```suggestion
 * @param {string} props.billingUrl The URL for setting the billing up in Google Ads
 * @param {Function} props.onSetupComplete Callback function when setup is completed
```

puntope commented on js/src/setup-mc/setup-stepper/saved-setup-stepper.js:
```diff
@@ -16,8 +16,13 @@ import StoreRequirements from './store-requirements';
-const SavedSetupStepper = ( props ) => {
-	const { savedStep, onRefetchSavedStep = () => {} } = props;
+/**
+ * @param {Object} props
+ * @param {string} [props.savedStep]
+ * @param {Function} [props.onRefetchSavedStep]
```
```suggestion
 * @param {Object} props React props
 * @param {string} [props.savedStep] A saved step overriding the current step
 * @param {Function} [props.onRefetchSavedStep] Callback when Saved Step is updated
```

puntope commented on js/src/setup-mc/setup-stepper/setup-accounts/google-mc-account/connected-card.js:
```diff
@@ -18,8 +18,18 @@ import useDispatchCoreNotices from '.~/hooks/useDispatchCoreNotices';
+/**
+ * @fires gla_mc_account_connect_different_account_button_click
+ * @param {Object} props React props
+ * @param {Object} props.googleMCAccount
```
```suggestion
 * @param {Object} props.googleMCAccount Connected Google Merchant Center Account
```

puntope commented on js/src/setup-mc/setup-stepper/setup-accounts/google-mc-account/reclaim-url-card/index.js:
```diff
@@ -23,8 +23,21 @@ import AccountCard, { APPEARANCE } from '.~/components/account-card';
+ * @param {Object} props React props
+ * @param {string} props.id
+ * @param {string} props.websiteUrl
+ * @param {Function} [props.onSwitchAccount]
```
```suggestion
 * @param {string} props.id Google Account ID
 * @param {string} props.websiteUrl Website's URL
 * @param {Function} [props.onSwitchAccount] Callback when clicking on Switch Account
```

puntope commented on js/src/setup-mc/setup-stepper/setup-accounts/google-mc-account/switch-url-card/index.js:
```diff
@@ -22,8 +22,27 @@ import AccountCard, { APPEARANCE } from '.~/components/account-card';
+
+/**
+ * @param {Object} props React props.
+ * @param {string} props.id
+ * @param {string} props.claimedUrl
+ * @param {string} props.newUrl
+ * @param {Function} [props.onSelectAnotherAccount]
```
```suggestion
 * @param {string} props.id Google Account ID
 * @param {string} props.claimedUrl The claimed URL 
 * @param {string} props.newUrl The new URL to be associated to the Google Account
 * @param {Function} [props.onSelectAnotherAccount] Callback when a different account is selected
```

puntope commented on js/src/setup-mc/setup-stepper/setup-accounts/google-mc-account/terms-modal/index.js:
```diff
@@ -13,8 +13,22 @@ import AppDocumentationLink from '.~/components/app-documentation-link';
+
+/**
+ * @param {Object} props React props
+ * @param {Function} [props.onCreateAccount]
+ * @param {Function} [props.onRequestClose]
```
```suggestion
 * @param {Function} [props.onCreateAccount] Callback function when the account is created
 * @param {Function} [props.onRequestClose] Callback function when the modal is closed
```

puntope commented on js/src/setup-mc/setup-stepper/setup-accounts/google-mc-account/warning-modal/index.js:
```diff
@@ -12,13 +12,25 @@ import AppButton from '.~/components/app-button';
 
+/**
+ * @param {Object} props React props.
+ * @param { { name: string, domain:string } } props.existingAccount
+ * @param {Function} [props.onContinue]
+ * @param {Function} [props.onRequestClose]
```
```suggestion
 * @param { { name: string, domain:string } } props.existingAccount Account name and domain 
 * @param {Function} [props.onContinue] Callback function when confirming the modal
 * @param {Function} [props.onRequestClose] Callback function when closing the modal
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -7,6 +7,22 @@ import { recordEvent } from '@woocommerce/tracks';
+ * Toggling display of table columns
+ *
+ * @event gla_table_header_toggle
+ * @property {string} report name of the report table (e.g. `"dashboard" | "reports-programs" | "reports-products" | "product-feed"`)
+ * @property {string} column name of the column
```
```suggestion
 * @property {string} report Name of the report table (e.g. `"dashboard" | "reports-programs" | "reports-products" | "product-feed"`)
 * @property {string} column Name of the column
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -7,6 +7,22 @@ import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ *
+ * @param {string} report
+ * @param {string} column
+ * @param {string} status
```
```suggestion
 * @param {string} report The report's name
 * @param {string} column Column that was toggled
 * @param {string} status (`on`|`off`) Indicates if the column was toggled on or off. 
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -15,10 +31,42 @@ export const recordTableHeaderToggleEvent = ( report, column, status ) => {
+ * Sorting table
+ *
+ * @event gla_table_sort
+ * @property {string} report name of the report table (e.g. `"dashboard" | "reports-programs" | "reports-products" | "product-feed"`)
+ * @property {string} column name of the column
+ * @property {string} direction (`asc`|`desc`)
```
```suggestion
 * @property {string} report Name of the report table (e.g. `"dashboard" | "reports-programs" | "reports-products" | "product-feed"`)
 * @property {string} column Name of the column
 * @property {string} direction (`asc`|`desc`)
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -15,10 +31,42 @@ export const recordTableHeaderToggleEvent = ( report, column, status ) => {
+
+/**
+ *
+ * @param {string} report
+ * @param {string} column
+ * @param {string} direction
```
```suggestion
 * @param {string} report The report's name
 * @param {string} column Column that was sorted
 * @param {string} direction (`asc`|`desc`) Indicates if it was sorted in ascending or descending order
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -15,10 +31,42 @@ export const recordTableHeaderToggleEvent = ( report, column, status ) => {
+ * When table pagination is changed by entering page via "Go to page" input.
+ *
+ * @event gla_table_go_to_page
+ * @property {string} context name of the table
+ * @property {string} page page number (starting at 1)
```
```suggestion
 * @property {string} context Name of the table
 * @property {string} page Page number (starting at 1)
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -15,10 +31,42 @@ export const recordTableHeaderToggleEvent = ( report, column, status ) => {
+ * When table pagination is clicked
+ *
+ * @event gla_table_page_click
+ * @property {string} context name of the table
+ * @property {string} direction direction of page to be changed. `("next" | "previous")`
```
```suggestion
 * @property {string} context Name of the table
 * @property {string} direction Direction of page to be changed. `("next" | "previous")`
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -42,6 +93,18 @@ export const recordTablePageEvent = ( context, page, direction ) => {
+ * with report name and data that comes from `DateRangeFilterPicker`'s `onRangeSelect` callback
+ *
+ * @event gla_datepicker_update
+ * @property {string} report name of the report (e.g. `"dashboard" | "reports-programs" | "reports-products" | "product-feed"`)
```
```suggestion
 * @property {string} report Name of the report (e.g. `"dashboard" | "reports-programs" | "reports-products" | "product-feed"`)
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -52,10 +115,22 @@ export const recordTablePageEvent = ( context, page, direction ) => {
+ * Triggered when changing products & variations filter.
+ *
+ * @event gla_filter
+ * @property {string} report name of the report (e.g. `"reports-products"`)
+ * @property {string} filter value of the filter (e.g. `"all" | "single-product" | "compare-products"`)
+ * @property {string | undefined} variationFilter value of the variation filter (e.g. `undefined | "single-variation" | "compare-variations"`)
```
```suggestion
 * @property {string} report Name of the report (e.g. `"reports-products"`)
 * @property {string} filter Value of the filter (e.g. `"all" | "single-product" | "compare-products"`)
 * @property {string | undefined} variationFilter Value of the variation filter (e.g. `undefined | "single-variation" | "compare-variations"`)
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -80,6 +157,19 @@ export const recordChartTabClickEvent = ( data ) => {
+ * Setup Merchant Center
+ *
+ * @event gla_setup_mc
+ * @property {string} target button ID
+ * @property {string} trigger action (e.g. `click`)
```
```suggestion
 * @property {string} target Button ID
 * @property {string} trigger Action (e.g. `click`)
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -80,6 +157,19 @@ export const recordChartTabClickEvent = ( data ) => {
+ */
+
+/**
+ * @param {string} target
+ * @param {string} [trigger='click']
```
```suggestion
 * @param {string} target Target button ID
 * @param {string} [trigger='click'] The trigger action
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -91,16 +181,39 @@ export const recordPreLaunchChecklistCompleteEvent = () => {
+ *
+ * @event gla_setup_ads
+ * @property {string} target button ID
+ * @property {string} trigger action (e.g. `click`)
```
```suggestion
 * @property {string} target Button ID
 * @property {string} trigger Action (e.g. `click`)
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -91,16 +181,39 @@ export const recordPreLaunchChecklistCompleteEvent = () => {
+ */
+
+/**
+ * @param {string} target
+ * @param {string} trigger
```
```suggestion
 * @param {string} target Target Button ID
 * @param {string} trigger The action trigger
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -91,16 +181,39 @@ export const recordPreLaunchChecklistCompleteEvent = () => {
+ * Triggered when the "Launch paid campaign" button is clicked to add a new paid campaign
+ *
+ * @event gla_launch_paid_campaign_button_click
+ * @property {string} audiences country codes of the paid campaign audience countries, e.g. `'US,JP,AU'`. This means the campaign is created with the multi-country targeting feature. Before this feature support, it was implemented as 'audience'.
+ * @property {string} budget daily average cost of the paid campaign
```
```suggestion
 * @property {string} audiences Country codes of the paid campaign audience countries, e.g. `'US,JP,AU'`. This means the campaign is created with the multi-country targeting feature. Before this feature support, it was implemented as 'audience'.
 * @property {string} budget Daily average cost of the paid campaign
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -112,3 +225,33 @@ export const recordLaunchPaidCampaignClickEvent = ( budget, audiences ) => {
+ * Clicking on a Google Merchant Center link.
+ *
+ * @event gla_google_mc_link_click
+ * @property {string} context indicate which page / module the link is in
+ * @property {string} href link's URL
```
```suggestion
 * @property {string} context Indicates which page / module the link is in
 * @property {string} href Link's URL
```

puntope commented on js/src/utils/recordEvent.js:
```diff
@@ -112,3 +225,33 @@ export const recordLaunchPaidCampaignClickEvent = ( budget, audiences ) => {
+ * A modal is closed.
+ *
+ * @event gla_modal_closed
+ * @property {string} context indicate which modal is closed
+ * @property {string} action indicate the modal is closed by what action (e.g. `maybe-later`|`dismiss` | `create-another-campaign`)
```
```suggestion
 * @property {string} context Indicates which modal is closed
 * @property {string} action Indicates the modal is closed by what action (e.g. `maybe-later`|`dismiss` | `create-another-campaign`)
```

puntope commented:
💅 ❓ 

There is still a warning 

```
WARNING: The @ignore tag does not permit a value; the value will be ignored. File: index.js, line: 1
WARNING: The @ignore tag does not permit a value; the value will be ignored. File: index.js, line: 42
```

When running the job... however since all our files are named index.js I didn't localize the file.

puntope reviewed (approved):
Great job refactoring the docs 🚀 

✅  LGTM

I left a few suggestions mainly about  the next:

- Missing props description 
- Delete description in @ type prop in order to avoid warnings
- In regards event docs I like it more when the description starts with Uppercase, is easier to read for me, otherwise, I mess the prop name with the description a bit.  Also, we maintain consistency with function docs in which we use uppercase for starting the description.
- Still one warining more that I couldn't identify

tomalec committed:
Add missing component's props docs,

Editorial changes.

Apply suggestions from code review

Co-authored-by: Miguel Pérez Pellicer <5908855+puntope@users.noreply.github.com>

tomalec committed:
Merge branch 'tweak/jsdoc-gen' into tweak/jsdoc-move

Regenerate tracking docs.

tomalec commented:
Thanks for all the suggestions! :) 

> I like it more when the description starts with Uppercase, is easier to read for me, otherwise, I mess the prop name with the description a bit.

Same here :)

tomalec committed:
Remove unnecesary parenthesis from FAQ event docs,

addresses https://github.com/woocommerce/google-listings-and-ads/pull/1401#discussion_r843614815

tomalec committed:
Fix whiteaspaces in code docs.

To make linter happy.
